Blog announcing inaugural EKS to AKS migration post (#4466)
Azure/AKS: blog/_posts/2024-08-07-eks-to-aks-series-part1.md

---
title: "Mastering the Move: EKS to AKS by Example"
description: "Learn how to migrate Amazon EKS workloads to Azure AKS using KEDA and Karpenter in this comprehensive guide."

> date: 2050-08-24

Reviewer: Update this date to something today or earlier. This is used to render the page. Ideally the file name and this date match

> date: 2050-08-24

Reviewer: This date needs to be some date now or in the past or the post won't render. Maybe change to 2024-08-15

> date: 2050-08-24

Author: ```suggestion
date: 2024-08-01
```

> date: 2050-08-24

Reviewer 2: Change the date to ensure it is publishes as soon as it is merged.

```suggestion
date: 2024-08-15
```

author: Kenneth Kilty
categories: general
---

## Introduction

> Many companies use multiple clouds for their workloads. These companies need to accommodate the cloud preferences of their customers. Kubernetes plays a central role in multi-cloud workloads due to its ability to provide a consistent and portable environment across different cloud providers.

Reviewer 3: Accommodating customer cloud preference is a driver for some, but not all customers who use multiple clouds.  "Some of these customers" is more accurate start for this sentence.

> Many companies use multiple clouds for their workloads. These companies need to accommodate the cloud preferences of their customers. Kubernetes plays a central role in multi-cloud workloads due to its ability to provide a consistent and portable environment across different cloud providers.

Author: ```suggestion
Many companies use multiple clouds for their workloads. Some of these companies need to accommodate the cloud preferences of their customers. Kubernetes plays a central role in multi-cloud workloads due to its ability to provide a consistent and portable environment across different cloud providers.
```

> We would like to share the first in a new documentation series designed specifically for customers already using Amazon EKS, to help them replicate or migrate their workloads to AKS: [Replicate an AWS event-driven workflow (EDW) workload with KEDA and Karpenter in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview)

Reviewer 2: ```suggestion
We would like to share the first in a new documentation series designed specifically for customers already using Amazon EKS, to help them replicate or migrate their workloads to AKS: [Replicate an AWS event-driven workflow (EDW) workload with KEDA and Karpenter in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/azure/aks/eks-edw-overview)
```

> We’re excited to announce the first in a new documentation series designed specifically for customers already using Amazon EKS, to help them replicate or migrate their workloads to AKS: [Replicate an AWS event-driven workflow (EDW) workload with KEDA and Karpenter in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview)

Reviewer 4: "We're excited to announce" always triggers my marketing detector and I really think we want to avoid that vibe in the blog. I'd suggest sticking to more neutral/fact-based language and let the reader decide whether it's exciting. :)

> We’re excited to announce the first in a new documentation series designed specifically for customers already using Amazon EKS, to help them replicate or migrate their workloads to AKS: [Replicate an AWS event-driven workflow (EDW) workload with KEDA and Karpenter in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview)

Author: ```suggestion
We would like to share the first in a new documentation series designed specifically for customers already using Amazon EKS, to help them replicate or migrate their workloads to AKS: [Replicate an AWS event-driven workflow (EDW) workload with KEDA and Karpenter in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview)
```

> Moving a workload from another managed Kubernetes offering to AKS can be challenging. This series aims to provide a holistic, end-to-end experience by example to help ease that process. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.

Reviewer 4: Rather than saying that moving a workload from another managed K8s offering can be challenging, I think it would be more accurate to say something like "Even with a portable API like Kubernetes offers, moving between clouds can be challenging". You could also throw in an explanatory metaphor here to make the post a little more informal - eg. "Every cloud has its own unique concepts and behaviors that can seem foreign when you're accustomed to another cloud's approach, not unlike the experience of learning a new language or visiting a new country for the first time. Think of us like your language tutor and local guide to the world of Azure."

> Moving a workload from another managed Kubernetes offering to AKS can be challenging. This series aims to provide a holistic, end-to-end experience by example to help ease that process. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.

Author: ```suggestion
Even with Kubernetes’ portable API, moving between clouds can be challenging. Each cloud has its own unique concepts, behaviors, and quirks that will seem unfamiliar when you’re accustomed to another cloud’s approach. This is not unlike the experience of learning a new language or visiting a new country for the first time. This series will be local guide to the world of Azure. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.
```

> Even with Kubernetes’ portable API, moving between clouds can be challenging. Each cloud has its own unique concepts, behaviors, and quirks that will seem unfamiliar when you’re accustomed to another cloud’s approach. This is not unlike the experience of learning a new language or visiting a new country for the first time. This series will be local guide to the world of Azure. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.

Reviewer 5: I would swap "quirks" for characteristics

> Even with Kubernetes’ portable API, moving between clouds can be challenging. Each cloud has its own unique concepts, behaviors, and quirks that will seem unfamiliar when you’re accustomed to another cloud’s approach. This is not unlike the experience of learning a new language or visiting a new country for the first time. This series will be local guide to the world of Azure. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.

Reviewer 5: Did you mean "This series will be a local guide to the world of Azure."

> Even with Kubernetes’ portable API, moving between clouds can be challenging. Each cloud has its own unique concepts, behaviors, and quirks that will seem unfamiliar when you’re accustomed to another cloud’s approach. This is not unlike the experience of learning a new language or visiting a new country for the first time. This series will be local guide to the world of Azure. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.

Author: ```suggestion
Even with Kubernetes’ portable API, moving between clouds can be challenging. Each cloud has its own unique concepts, behaviors, and characteristics that will seem unfamiliar when you’re accustomed to another cloud’s approach. This is not unlike the experience of learning a new language or visiting a new country for the first time. This series will be a local guide to the world of Azure. The samples in this series begin with infrastructure and code on EKS and end with equivalently functional infrastructure and code on AKS, while explaining the conceptual differences between AWS and Azure throughout.
```

## EKS to AKS - Event Driven Workflow Scenario

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an Amazon Web Services (AWS) Elastic Kubernetes Service (EKS) event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Reviewer 4: You're spelling out what EKS means here but you already used the acronym above. Suggest moving the full name to the first place you use it in the article.

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an Amazon Web Services (AWS) Elastic Kubernetes Service (EKS) event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Reviewer 4: On the point about minimizing changes - it sounds like there's an implicit tradeoff there that you might want to call out.  Is there some way in which the architecture could be even better if you leaned into azure fully? That could be a good thing to mention as it suggests that there's a spectrum of portability to value that the user can move along based on their goals.

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an Amazon Web Services (AWS) Elastic Kubernetes Service (EKS) event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Author: ```suggestion
This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.
```

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Reviewer 5: I believe this sentence needs a "of" - In the spirit of keeping the workload as portable as possible

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Author: ```suggestion
This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit of keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.
```

> This new [article](https://learn.microsoft.com/en-us/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.

Reviewer 2: ```suggestion
This new [article](https://learn.microsoft.com/azure/aks/eks-edw-overview) guides you through replicating an EKS event-driven workflow (EDW) workload using [KEDA](https://keda.sh/) and [Karpenter](https://karpenter.sh/) in Azure Kubernetes Service (AKS). In the spirit keeping the workload as portable as possible, this guide prioritizes minimizing the changes needed to get the workload running on Azure.
```

> In this first article, the workload implements the competing consumers pattern with a producer/consumer app, facilitating efficient data processing by separating data production from data consumption. KEDA is used to scale pods running consumer processing, while AKS managed Karpenter autoscale Kubernetes nodes through the new AKS [Node Autoprovisioning (NAP)](https://learn.microsoft.com/en-gb/azure/aks/node-autoprovision) feature currently available in preview.

Reviewer 2: ```suggestion
The sample workload implements the competing consumers pattern with a producer/consumer app, facilitating efficient data processing by separating data production from data consumption. KEDA is used to scale pods running consumer processing, while AKS managed Karpenter autoscale Kubernetes nodes through the new AKS [Node Autoprovisioning (NAP)](https://learn.microsoft.com/azure/aks/node-autoprovision) feature currently available in preview.
```

You can either clone the AWS EKS sample and follow along with the guide, or read the guide to understand the necessary changes and the reasons behind them as outlined below.

1. Understand Platform Differences: Review the differences between EKS and AKS in terms of services, architecture, and deployment for this workload.
2. Re-architect the Workload: Analyze the existing EKS workload architecture and identify components that need changes to fit AKS.
3. Update Application Code: Refactor elements of the code to make the app compatible with Azure APIs, services, and authentication models.
4. Prepare for Deployment: Modify the EKS deployment process to use the Azure CLI.
5. Deploy the Workload: Deploy the replicated EKS workload in AKS and test to ensure it functions as expected.

> ## Summary

Author: ```suggestion
## Looking forward
```

> ## Summary

Reviewer 2: ```suggestion
## Looking ahead
```

> Learn how to migrate Amazon EKS workloads to Azure AKS using KEDA and Karpenter in this comprehensive guide. Gain insights into the platform differences, re-architect the workload, update application code, prepare for deployment, and deploy the workload in AKS.

Reviewer 4: This is kind of a dry ending and I don't think the blog post is long enough to really need a recap. Can we close with something more forward looking, like about what additional guides are coming? Maybe encourage people to reach out somehow with other ideas?

> Learn how to migrate Amazon EKS workloads to Azure AKS using KEDA and Karpenter in this comprehensive guide. Gain insights into the platform differences, re-architect the workload, update application code, prepare for deployment, and deploy the workload in AKS.

Author: ```suggestion
Our next article will explore a common EKS workload pattern for protecting an n-tier web/mobile application with a WAF, and the process of replicating this workload into Azure using EKS and other services. We are interested to hear about the migration challenges and scenarios you encounter, so we can continue to enhance the features and documentation for AKS to ease the transition from other clouds to Azure. Please share your ideas and needs using one of the contact methods below.
```
